dotnet/src/Connectors/Connectors.OpenAIV2/OpenAIAudioToTextExecutionSettings.cs

```diff
@@ -0,0 +1,190 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Execution settings for OpenAI audio-to-text request.
+/// </summary>
+[Experimental("SKEXP0010")]
+public sealed class OpenAIAudioToTextExecutionSettings : PromptExecutionSettings
+{
+    /// <summary>
+    /// Filename or identifier associated with audio data.
+    /// Should be in format {filename}.{extension}
+    /// </summary>
+    [JsonPropertyName("filename")]
+    public string Filename
+    {
+        get => this._filename;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._filename = value;
+        }
+    }
+
+    /// <summary>
+    /// An optional language of the audio data as two-letter ISO-639-1 language code (e.g. 'en' or 'es').
+    /// </summary>
+    [JsonPropertyName("language")]
+    public string? Language
+    {
+        get => this._language;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._language = value;
+        }
+    }
+
+    /// <summary>
+    /// An optional text to guide the model's style or continue a previous audio segment. The prompt should match the audio language.
+    /// </summary>
+    [JsonPropertyName("prompt")]
+    public string? Prompt
+    {
+        get => this._prompt;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._prompt = value;
+        }
+    }
+
+    /// <summary>
+    /// The format of the transcript output, in one of these options: json, text, srt, verbose_json, or vtt. Default is 'json'.
+    /// </summary>
+    [JsonPropertyName("response_format")]
+    public string ResponseFormat
+    {
+        get => this._responseFormat;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._responseFormat = value;
+        }
+    }
+
+    /// <summary>
+    /// The sampling temperature, between 0 and 1.
+    /// Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.
+    /// If set to 0, the model will use log probability to automatically increase the temperature until certain thresholds are hit.
+    /// Default is 0.
+    /// </summary>
+    [JsonPropertyName("temperature")]
+    public float Temperature
+    {
+        get => this._temperature;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._temperature = value;
+        }
+    }
+
+    /// <summary>
+    /// The timestamp granularities to populate for this transcription. response_format must be set verbose_json to use timestamp granularities. Either or both of these options are supported: word, or segment.
+    /// </summary>
+    [JsonPropertyName("granularities")]
+    public IReadOnlyList<TimeStampGranularities>? Granularities { get; set; }
```

Comment:
Is it necessary to supply more than one granularity? It seems the underlying API supports only one at a time.

Reply:
Originally the Granularities are multiple options, but for some reason they are only supporting one at a time, I will update our settings to accept only one as we did change our settings in `ChatCOmpletion`

Created a Issue in OpenAI repo to track this: 
https://github.com/openai/openai-dotnet/issues/104

https://platform.openai.com/docs/api-reference/audio/createTranscription#audio-createtranscription-timestamp_granularities